sycl/include/sycl/detail/defines_elementary.hpp

```diff
@@ -58,6 +58,20 @@
 #endif
 #endif // __SYCL2020_DEPRECATED
 
+#ifndef __SYCL_WARN_IMAGE_ASPECT
+#if !defined(SYCL_DISABLE_IMAGE_ASPECT_WARNING) && __has_attribute(diagnose_if)
+#define __SYCL_WARN_IMAGE_ASPECT(aspect_param)                                   \
+  __attribute__((diagnose_if(                                                    \
+      aspect_param == aspect::image,                                             \
```

Comment:
What happens here if the value of `aspect_param` is not known at compile time?

Reply:
I think that's OK.  I was worried that the compiler might diagnose an error if the expression `aspect_param == aspect::image` cannot be evaluated at compile time.